### PR TITLE
WIP: Revamp Components

### PR DESCRIPTION
**Pull request by lionize**
# Changes
- Update to webpack 2 
- Re-write component importing (see Imports)
- Remove css imports from webpack and components in preparation for style-components #75
- Install nyc and istanbul for code coverage #74
- Create `test:coverage` npm script #74
- Rename `utils/renderField.js` to `utils/render-field.js` to follow my own file naming standard
- Modify `TicketDeleteButton` to take ticket prop #118
- Test Components #107
- Test Containers #116 

# Details
## Imports
Each folder in `components/` has an `index.js` that exports all components in that folder. `components/index.js` then imports the components from each folder and re-exports them. This allows us to import many components from different folders in a single statement, e.g. `import { Button, ProjectShow, UserList } from 'components'`.

When creating new components, follow these steps:
1. If component is in a new folder, create the `index.js` file for that folder.
2. Re-export the component in its folder's `index.js`:
```js
export { default as ComponentName } from 'components/subfolder/component-name'
```
3. If this is a new folder, import the folder in `components/index.js`:
```js
import * as newComponents from 'components/subfolder'

const containers = [
{...}
  newComponents,
]
```

## Component Tests

Completes #107. Each presentational component has basic tests that cover callback functionality and dynamic rendered information.

## Container  Tests

Completes #116.  Each container component now tests that it renders, that it passes the correct callbacks to its child, and that the child calls the callbacks correctly.

## Test Utility HOCs

Created `wrapRouter` and `wrapProvider` HOCs for testing purposes. These allow us to mock the `react-redux` `Provider` for components that use `connect` and the `react-router` router for components that use `withRouter`.

An example from `tests/components/TicketEdit.spec.js`:

```javascript
const state = {
  data: {
    tickets: {
      keys: ['TICKET1'],
      byKey: {
        TICKET1: {
          key: 'TICKET1'
        }
      }
    }
  }
}
const params = {
  key: 'TICKET1'
}

let Enhanced = wrapProvider({ state })(Container)
Enhanced = wrapRouter({ params })(Enhanced)

const wrapper = mount(<Enhanced />)
```

The `Enhanced` component now has the router parameters and other information injected into its context and the state injected into a `Provided` component which is now outermost component.

### CODE REVIEW

**Comment by lionize** [comment]
@chasinglogic Could you take a look at my travis.yml? The lint command is exiting with 0 due to errors but it's still showing the tests as passing.

File: mochacfg.js [hunk 1]
```diff
@@ -0,0 +1 @@
+require.extensions['.png'] = function mochaPngConfig() { return null; }
```
Comment (chasinglogic) [inline mochacfg.js]: I would add a comment explaining what this does, because without googling I have no idea.

**Comment by chasinglogic** [comment]
We really need a HACKING.md and / or CONTRIBUTING.md doc for the frontend.

File: mochacfg.js (hunk 1 above)
Comment (lionize) [inline mochacfg.js]: I'm not entirely sure I know what it does lol, it just makes the coverage script work. I'll add that note though.

**Comment by lionize** [comment]
I agree with that. I'm planning on getting that together soon. Basic overview of project and its parts, what they do, etc.

**Comment by lionize** [comment]
Also a little section on testing.

**Comment by chasinglogic** [comment]
I would also recommend documenting what dependencies we have, what they do, and why we use them.

**Comment by lionize** [comment]
Also a good idea.

**Pull request merged**